### PR TITLE
[pt1][quant] Enable UBSAN test for FBGEMM in dynamic quant test

### PR DESCRIPTION
**Pull request by jianyuh**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25099 [pt1][quant] Enable UBSAN test for FBGEMM in dynamic quant test**

As title says

Differential Revision: [D16986248](https://our.internmc.facebook.com/intern/diff/D16986248/)

### CODE REVIEW

File: test/test_quantized.py [hunk 1]
```diff
@@ -532,10 +532,9 @@ def equal_ref(X, params, X_scheme, X2, params2, X2_scheme):
 
 
 @unittest.skipIf(
-    TEST_WITH_UBSAN or not torch.fbgemm_is_cpu_supported(),
-    " Quantized Linear requires FBGEMM. FBGEMM does not play"
-    " well with UBSAN at the moment, so we skip the test if"
-    " we are in a UBSAN environment.",
+    not torch.fbgemm_is_cpu_supported(),
```
Comment (dskhudia) [inline test/test_quantized.py]: Remove TEST_WITH_UBSAN import at the top.

**Review by dskhudia — APPROVED** [review]

File: test/test_quantized.py (hunk 1 above)
Comment (jianyuh) [reply test/test_quantized.py]: QNNPACK test also used `TEST_WITH_UBSAN` to disable UBSAN test. Let me check if we can also remove that.

Comment (supriyar) [reply test/test_quantized.py]: We don't want to remove it for QNNPACK just yet. Some kernels still fail with ASAN and have to be fixed.

Comment (jianyuh) [reply test/test_quantized.py]: Will keep the `TEST_WITH_UBSAN` for QNNPACK

**Pull request closed without merging**